toolchain/src/integration-test/java/org/archcnl/toolchain/ToolchainIT.java

```diff
@@ -0,0 +1,92 @@
+package org.archcnl.toolchain;
+
+import static org.junit.Assert.assertTrue;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+
+import org.apache.jena.ontology.OntModel;
+import org.apache.jena.ontology.OntModelSpec;
+import org.apache.jena.query.Query;
+import org.apache.jena.query.QueryExecution;
+import org.apache.jena.query.QueryExecutionFactory;
+import org.apache.jena.query.QueryFactory;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.archcnl.stardogwrapper.api.StardogDatabaseAPI;
+import org.archcnl.stardogwrapper.impl.StardogDatabase;
+import org.junit.Test;
+
+public class ToolchainIT {
+
+    private final String rootDir = "./src/integration-test/resources/";
+    private final String database = "archcnl_it_db";
+    private final String server = "http://localhost:5820";
+    private final String username = "admin";
+    private final String password = "admin";
+    private final String context = "http://graphs.org/" + database + "/1.0";
+    private final String resultPath = rootDir + "result.owl";
+    
+    @Test
+    public void testArchcnl() throws IOException {
+        CNLToolchain.runToolchain(database,
+                server,
+                context,
+                username,
+                password,
+                Arrays.asList(rootDir + "OnionArchitectureDemo/src/"), // source paths
+                rootDir + "OnionArchitectureDemo/rules.adoc", // rule file 
+                false, // verbose logging
+                true); // remove previous databases
+        
+        
+        writeResultToFile();
+        
+        OntModel result = loadResult();
+
+        assertTrue(isFirstRuleViolated(result));
+        assertTrue(isSecondRuleViolated(result));
+    }
+
+    private OntModel loadResult() throws IOException {
+        // Note: Stardog's context is the name of the RDF graph in which the ontology
+        // is stored. During the writing from DB/loading into RAM steps, this database name
+        // is lost. All data stored in result is an unnamed graph.
+        // Thus, "GRAPH ?g {}" must not be included in the SPARQL queries executed on result.
+        // WHERE {GRAPH ?g { ... }} works only on named graphs, while WHERE { ... } works only
+        // on the default graph. Thus, the queries in the SPARQL-queries directory (which are suppposed to
+        // be executed on the DB directly) use GRAPH and the queries used by this test do not.
+        OntModel result = ModelFactory.createOntologyModel(OntModelSpec.OWL_MEM, null);
+        result.read(resultPath);
+        result.write(new FileWriter(rootDir + "input.ttl"), "TURTLE");
```

Comment:
Why is the call to `write()` necessary? Who will read this file?

Reply:
It's a leftover from debugging. I will remove it.

Reply:
:+1: